Sources/TuistCore/Utils/OtoolController.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistSupport
+
+enum OtoolErrors: FatalError, Equatable {
+    case invalidOutput
+
+    var description: String {
+        switch self {
+        case .invalidOutput: return "The output is not in the format tuist expects"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .invalidOutput: return .abort
+        }
+    }
+}
+
+/// Otool controller protocol passed in initializers for dependency injection
+public protocol OtoolControlling {
+    /// Returns the paths to dlyb dependencies.
+    /// - Parameter path: Path to the given dependency
+    /// - Returns Array of dependencies returned by the command.
+    func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]>
+}
+
+/// OtoolController
+/// Used to find out against which dynamic libraries a certain binary is linked.
+public struct OtoolController: OtoolControlling {
```

Comment:
This one is a good candidate for an integration test using the fixtures that you added in this PR. I'd test that given a binary path, I obtain the right dlyb paths using the method in this utility. Thanks to this, if the contract with `otool` breaks, our tests will fail and we'll catch it before it gets into the users' hands.